File: frontend/src/api/category.ts
import request from './index'

// 分类类型：收入/支出
export type CategoryType = 'INCOME' | 'EXPENSE'

export interface Category {
  category_id: string
  name: string
  type: CategoryType
  description: string
  sort_order: number
  is_deleted: number
  create_time: string
  update_time: string
}

export interface CreateCategoryParams {
  name: string
  type: CategoryType
  description?: string
  sort_order?: number
}

// 创建费用分类
export function createCategory(data: CreateCategoryParams) {
  return request.post('/categories', data)
}

// 获取分类列表
export function getCategoryList(params?: { page?: number; page_size?: number; keyword?: string; type?: CategoryType }) {
  return request.get('/categories', { params })
}

// 更新分类
export function updateCategory(id: string, data: Partial<Category>) {
  return request.put(`/categories/${id}`, data)
}

// 删除分类
export function deleteCategory(id: string) {
  return request.delete(`/categories/${id}`)
}

File: backend/models/category.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Category 费用分类模型
type Category struct {
	CategoryID   string         `gorm:"column:category_id;primaryKey;size:32" json:"category_id"`
	Name         string         `gorm:"column:name;uniqueIndex;size:50;not null" json:"name"`
	Description  string         `gorm:"column:description;size:200" json:"description"`
	SortOrder    int            `gorm:"column:sort_order;default:0" json:"sort_order"`
	IsDeleted    int            `gorm:"column:is_deleted;default:0" json:"is_deleted"` // 0-正常，1-已删除
	CreateTime   time.Time      `gorm:"column:create_time;autoCreateTime" json:"create_time"`
	UpdateTime   time.Time      `gorm:"column:update_time;autoUpdateTime" json:"update_time"`
	DeleteTime   gorm.DeletedAt `gorm:"column:delete_time;index" json:"-"`
}

// TableName 指定表名
func (Category) TableName() string {
	return "t_category"
}
